runtime/near-vm-errors/src/lib.rs

```diff
@@ -291,22 +292,21 @@ impl From<InconsistentStateError> for VMLogicError {
     }
 }
 
-impl From<PrepareError> for VMError {
+impl From<PrepareError> for FunctionCallError {
     fn from(err: PrepareError) -> Self {
-        VMError::FunctionCallError(FunctionCallError::CompilationError(
-            CompilationError::PrepareError(err),
-        ))
+        FunctionCallError::CompilationError(CompilationError::PrepareError(err))
     }
 }
 
-impl From<VMLogicError> for VMError {
-    fn from(err: VMLogicError) -> Self {
+impl TryFrom<VMLogicError> for FunctionCallError {
```

Comment:
I think this impl is load bearing, could you write a doc comment explaining the classification here (which error is ok, and which is err)